### PR TITLE
DWW OKTA-186278 doc Linked Object property name restrictions

### PR DESCRIPTION
**Pull request by daytonwilliams-okta**
## Description:
doc Linked Object property name restrictions

### Resolves:
* [OKTA-186278](#186278)

* [OKTA-186278](https://oktainc.atlassian.net/browse/OKTA-186278)

### CODE REVIEW

**Comment by daytonwilliams-okta** [comment]
Merging to `source` because it is fine if this doc update precedes the associated code change by a short while.

1) The change was merged last Friday, so it is already on a train

2) Because of the peculiar characteristics of this fix, the doc update is actually mostly independent of the code change.

Feature is EA, which helps.

Bug:  Hey, I did X and now this feature doesn't work
Analysis: Doing X should never have been allowed because it corrupts the database.

Code fix: Add validation to prevent people from doing X.
Doc change: Thou shalt not do X.

So there's no reason not to tell people not to do it right away.

File: _source/_docs/api/resources/linked-objects.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Each org can create up to 200 definitions, and assign them to an unlimited numbe
 
 {% api_operation post /api/v1/meta/schemas/user/default/linkedObjects %}
 
-Adds a linked object definition to the user profile schema
+Adds a linked object definition to the user profile schema.  The `primary` and `associated` `name` fields may not start with a digit or contain anything other than `a-z`, `A-Z`, `0-9`, and `_` characters.
```
Comment (donwoods-okta) [inline _source/_docs/api/resources/linked-objects.md]: "digit or" ---> "digit, nor"

I also think the last part might read more smoothly if rearranged:
nor contain any characters other than `a-z`, `A-Z`, `0-9`, and `_`.

Comment (daytonwilliams-okta) [reply _source/_docs/api/resources/linked-objects.md]: I actually think it's more readable this way.  I will defer to the expertise of the tech writers who are supposed to review this.

Comment (jakubvul) [reply _source/_docs/api/resources/linked-objects.md]: Disagree about "nor", which is pretty much only used with "neither".

But I would rephrase it as such: 

> The `name` field found in both the `primary` and `associated` objects can only contain the following characters: `a-z`, `A-Z`, `0-9`, and `_`.

Comment (daytonwilliams-okta) [reply _source/_docs/api/resources/linked-objects.md]: @jakubvul That change would write out the other half of the restriction ("may not start with a digit").  How about:

> The `name` field found in both the `primary` and`associated` objects may not start with a digit, and can only contain the following characters: `a-z`, `A-Z`, `0-9`, and `_`.

Comment (jakubvul) [reply _source/_docs/api/resources/linked-objects.md]: Sorry, missed that part!

Yes I like your revision.

Comment (donwoods-okta) [reply _source/_docs/api/resources/linked-objects.md]: "Nor" is not limited to use with "neither".  Nor is "or" limited to use with "either".  (As an example.)  However, I'm fine with Dayton's rewording (recapturing the briefly lost leading-digit restriction).
